envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/filters/RBACFilterFactoryTest.kt

```diff
@@ -0,0 +1,71 @@
+package pl.allegro.tech.servicemesh.envoycontrol.snapshot.listeners.filters
+
+import io.envoyproxy.envoy.api.v2.route.HeaderMatcher
+import io.envoyproxy.envoy.config.rbac.v2.Permission
+import io.envoyproxy.envoy.config.rbac.v2.Policy
+import io.envoyproxy.envoy.config.rbac.v2.Principal
+import io.envoyproxy.envoy.config.rbac.v2.RBAC
+import org.junit.jupiter.api.Test
+import pl.allegro.tech.servicemesh.envoycontrol.groups.Incoming
+import pl.allegro.tech.servicemesh.envoycontrol.groups.IncomingEndpoint
+import pl.allegro.tech.servicemesh.envoycontrol.groups.PathMatchingType
+import pl.allegro.tech.servicemesh.envoycontrol.snapshot.IncomingPermissionsProperties
+import org.assertj.core.api.Assertions.assertThat
+
+internal class RBACFilterFactoryTest {
+    val rbacFilterFactory = RBACFilterFactory(IncomingPermissionsProperties())
+
+    val expected = RBAC.newBuilder().putAllPolicies(mapOf(
+            "client1,client2" to Policy.newBuilder().addAllPrincipals(listOf(
```

Comment:
Wouldn't it be more readable, if we write expected RBAC config in JSON/YAML? 
I think it would be easier to decide, if this config makes sense. 

Converting JSON to Protobuf object should be easy using `JsonFormat.parser()`